### PR TITLE
w:blobSoftRefField

### PR DESCRIPTION
**Pull request by jmuscireum**

### CODE REVIEW

File: src/main/resources/default/extensions/wondergem-page/storage-vfs-modals.html.pasta [hunk 1]
```diff
@@ -105,7 +105,7 @@
         </w:modal>
 
         <script type="text/javascript">
-            function selectVFSFile(initialPath) {
+            function selectVFSFile(initialPath, restictToPath) {
```
Comment (idlira) [inline src/main/resources/default/extensions/wondergem-page/storage-vfs-modals.html.pasta:108]:
```suggestion
            function selectVFSFile(initialPath, restrictToPath) {
```

File: src/main/resources/default/taglib/w/blobSoftRefField.html.pasta [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            selectVFSFile('', blobStorageSpacePath).then(function (selectedValue) {
+                $.getJSON('/dasd/blob-info-for-path/___blobRef.getSpace()', {
+                    path: selectedValue.replace(blobStorageSpacePath, '')
+                }, function (json) {
+                    alert(json.fileId);
+                });
+            });
+        });
+    });
+</script>
```
Comment (idlira) [inline src/main/resources/default/taglib/w/blobSoftRefField.html.pasta:74]: missing EOL

File: src/main/resources/default/extensions/wondergem-page/storage-vfs-modals.html.pasta (hunk 1 above)
Comment (idlira) [inline src/main/resources/default/extensions/wondergem-page/storage-vfs-modals.html.pasta:108]: filefield.html.pasta uses it at well.
Code using restrictToPath still in progress or not yet pushed?

**Review by idlira — CHANGES_REQUESTED** [review]
...

File: src/main/java/sirius/biz/storage/layer2/BlobController.java [hunk 3]
```diff
@@ -86,4 +85,25 @@ public void uploadFile(final WebContext ctx,
...
+                                                                    "Cannot find blob for path '%s' in space '%s'",
+                                                                    path,
+                                                                    spaceName)
+                                                            .handle());
+
+        out.property("fileId", blob.getBlobKey());
+        out.property("filename", blob.getFilename());
+        out.property("size", blob.getSize());
+        out.property("formattedSize", NLS.formatSize(blob.getSize()));
+        out.property("downloadUrl", blob.url().asDownload().buildURL().orElse(""));
```
Comment (andyHa) [inline src/main/java/sirius/biz/storage/layer2/BlobController.java:107]: isn't this a security issue? as "anyone" (who is REALLY GOOD as guessing ids) could obtain a download link - coudln't we handle this on another layer?

Comment (andyHa) [reply src/main/java/sirius/biz/storage/layer2/BlobController.java:107]: or maybe this is all good and properly enforced - just a random thought ;-)

**Pull request closed without merging**